cmd/demo/opasupport/opa_support_test.go

```diff
@@ -78,17 +82,58 @@ func TestMiddleware(t *testing.T) {
 	mockClient.response = []byte("{\"result\":true}")
 	server := setupWithMockClient(mockClient)
 
-	resp, err := http.Get(fmt.Sprintf("http://%s/", server.Addr))
+	resp, err := http.Post(fmt.Sprintf("http://%s/", server.Addr), "", nil)
 	if err != nil {
 		log.Fatalln(err)
 	}
+	defer resp.Body.Close()
 	body, _ := io.ReadAll(resp.Body)
 	assert.Equal(t, "opa!", string(body))
 	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	var actualOpaQuery opasupport.OpaQuery
+	err = json.Unmarshal(mockClient.requestBody, &actualOpaQuery)
+	assert.NoError(t, err)
+	assert.Equal(t, "http:POST", actualOpaQuery.Input["method"])
 
 	websupport.Stop(server)
 }
 
+func TestMiddlewareOpaQueryMethod(t *testing.T) {
+	mockClient := new(MockClient)
+	mockClient.response = []byte("{\"result\":true}")
+	support := opasupport.NewOpaSupport(mockClient, "aUrl", unauthorized)
+
+	handler := support.Middleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Calling the `support.Middleware` directly so I have more control over the request url. Since we use a real http server in the other tests, we could only send it urls with `http`. Open to better ways to test this behavior.